### PR TITLE
Add installation instructions

### PR DESCRIPTION
**Pull request by Vurp**
I think that it is important that build instructions is part of the repo itself. I had some issues with the instructions on the wiki, and had more success with [these](https://forum.zdoom.org/viewtopic.php?t=67477#p1137774) instructions. I added them to their own file and added what was missing from the wiki.

### CODE REVIEW

File: INSTALL [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	$ mkdir build
+	$ cd build
+	$ cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=`pwd`/../build_install ..
+
+2.  Build GZDoom.
+
+    If you acquired GZDOOM through git you may want to checkout the
+    stable version, which as of time of writing is 4.5.0:
+
+	$ git checkout g4.5.0
```
Comment (alexey-lysiuk) [inline INSTALL:59]: Building particular versions of ZMusic and GZDoom is questionable proposition. If you’re dealing with this yourself, you probably want the latest version, not the stable one.

File: INSTALL [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+1a. Make sure ZMusic is installed, either through your package manager
+    or by building it yourself.
+
+	$ git clone --depth 1 -b 1.1.6 https://github.com/coelckers/ZMusic.git
+	$ cd ZMusic
+	$ mkdir build
+	$ cd build
+	$ cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=`pwd`/../build_install ..
+
```
Comment (alexey-lysiuk) [inline INSTALL:53]: There is no actual build and install command.

File: INSTALL [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    If you built ZMusic yourself, set CMAKE_PREFIX_PATH to the full
+    path of the build_install directory set above, otherwise remove
+    the option entirely.
+
+	$ cd GZDoom
+	$ mkdir build
+	$ cd build
+	$ cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=../ZMusic/build/build_install/ ..
+	$ make
```
Comment (alexey-lysiuk) [inline INSTALL:69]: This command will build on one CPU core only if `MAKEFLAGS` environment variable isn’t set with `-j`/`--jobs` option.

**Review by alexey-lysiuk — CHANGES_REQUESTED** [review]

**Comment by madame-rachelle** [comment]
I am seconding everything Alexey said here - while I agree having install instructions within the repo would definitely be helpful, they should at least be correct, and Alexey helpfully pointed out all the errors.

File: INSTALL (hunk 2 above)
Comment (Vurp) [reply INSTALL:53]: I dont understand. No build or install command is referenced?

**Comment by Vurp** [comment]
There are still a lot of optional dependencies mentioned from the wiki
page I copied. How important are they? Do they still have a purpose?
The ones I am curios about are:

- git 

If you didn't acquire the source from cloning the repo you probably
dont need git either.

- zlib (GZDoom has a copy of it and will be statically compiled
  in if not found)
- libbzip2 (possibly static)
- libjpeg (possibly static)

As these are provided with the source, do they need to be mentioned?

- libgme or game-music-emu (possibly static)

After a cursory check of the repo it does not seem to be referenced.
Is this still used?

**Comment by alexey-lysiuk** [comment]
Git isn’t required but highly recommended way of obtaining source code. Versioning doesn’t work without it.

Installation of zlib, bzip2, jpeg is not required as well. However, some of them can be replaced with more optimized forks like zlib-ng and jpeg-turbo.

System’s GME cannot be used because every new version of it breaks something. It’s a ZMusic dependency anyway.

File: INSTALL (hunk 2 above)
Comment (alexey-lysiuk) [reply INSTALL:50]: Added `make` command still isn’t enough, and `-j` without number of jobs is really bad idea. It should be something like `make -j$(nproc) install`.

File: INSTALL [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    If you built ZMusic yourself, set CMAKE_PREFIX_PATH to the full
+    path of the build_install directory set above, otherwise remove
+    the option entirely.
+
+	$ cd GZDoom
+	$ mkdir build
+	$ cd build
+	$ cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=../ZMusic/build/build_install/ ..
+	$ make -j
```
Comment (alexey-lysiuk) [inline INSTALL:61]: `-j` alone is bad, it must be an explicit number of jobs specified.

**Comment by Vurp** [comment]
> Git isn’t required but highly recommended way of obtaining source code. Versioning doesn’t work without it.
> 
> Installation of zlib, bzip2, jpeg is not required as well. However, some of them can be replaced with more optimized forks like zlib-ng and jpeg-turbo.
> 
> System’s GME cannot be used because every new version of it breaks something. It’s a ZMusic dependency anyway.

Sounds like I should just leave the deps as they are. 

> `-j` alone is bad, it must be an explicit number of jobs specified.

Ok. I set a explicit number and added a explanation of what that does.
That should be the last issue resolved. This should be ready to be
merged.

File: INSTALL [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    Recommended
+      - GTK3 or GTK2
+      - git (needed in order to download the source and compile in
+        commit meta data)
+    Optional
+      - zlib (GZDoom has a copy of it and will be statically compiled
+        in if not found)
+      - libbzip2 (possibly static)
+      - libjpeg (possibly static)
+      - libgme or game-music-emu (possibly static)
```
Comment (alexey-lysiuk) [inline INSTALL:27]: Like I said, it's useless.

File: INSTALL [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      - FMOD Ex 4.36.x or later
```
Comment (alexey-lysiuk) [inline INSTALL:28]: Same here, it's useless as well.

File: INSTALL [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      - gxmessage (optional - needed to show the crash log in a
+        window)
+      - kdialog (optional - for KDE users)
+      - fluidsynth (optional - for MIDI playback)
+
+    In addition to or instead of FMOD, OpenAL can be used for sound in
```
Comment (alexey-lysiuk) [inline INSTALL:34]: This statement is wrong, there is no FMOD sound backend in GZDoom.

File: INSTALL (hunk 2 above)
Comment (alexey-lysiuk) [reply INSTALL:50]: Install rule is still missing, and hardcoded number of jobs isn't the best choice either.

Please use either tabs or spaces for indent, not a random mix of them.

File: INSTALL [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    path of the build_install directory set above, otherwise remove
+    the option entirely. The "-j 4" flag sets how many jobs to run
+    simultaneously. Set it too low and it will take needlessly long
+    time to build. Set it too high and you might run out of RAM.
+
+	$ cd GZDoom
+	$ mkdir build
+	$ cd build
+	$ cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=../ZMusic/build/build_install/ ..
+	$ make -j 4
```
Comment (alexey-lysiuk) [inline INSTALL:63]: Hardcoded number of jobs again.

File: INSTALL [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+1.  Install the required dependencies. Depending on your distro they
+    may have different package names. These are:
+
+    Required
+      - gcc 7.x or later
+      - make
+      - cmake 3.1 or later
+      - SDL 2.0.6 or later
+      - libGL and libGLU (SDL or libSDL pull in libGLU) or any other
```
Comment (alexey-lysiuk) [inline INSTALL:15]: OpenGL implementation should be pulled by SDL, and not installed directly.

**Review by alexey-lysiuk — CHANGES_REQUESTED** [review]
Don't take it personally, but an overall impression is like you don't clearly understand what you are trying to describe.
Some steps worked for you somehow, and you decided to write a guide. That's OK, but first you should try to do it from scratch, without lots of useless actions. Instead, you put everything you tried to this guide.

**Comment by coelckers** [comment]
Unresolved and dead for a year, off this goes.

**Pull request closed without merging**